### PR TITLE
include push and pull status code in beginsyncresponse.syncinfo

### PR DESCRIPTION
**Pull request by phritz**
progress towards https://github.com/rocicorp/replicache-sdk-js/issues/106

it's not perfect, but it solves the proximate issue described in that bug.

### CODE REVIEW

File: src/sync/mod.rs [hunk 1]
```diff
@@ -65,22 +68,41 @@ pub async fn begin_sync(
...
+            Ok(_) => Some(BatchPushInfo {
+                http_status_code: 200,
+                error_message: str!(""),
+            }),
+            Err(PushError::FetchNotOk(status_code)) => Some(BatchPushInfo {
+                http_status_code: u16::from(status_code),
+                error_message: format!("{:?}", PullError::FetchNotOk(status_code)),
+            }),
+            Err(e) => Some(BatchPushInfo {
+                http_status_code: 0, // TOOD we could return this properly in the PushError.
```
Comment (aboodman) [inline src/sync/mod.rs:102]: TODO

Comment (aboodman) [reply src/sync/mod.rs:102]: I just fixed it for you :)

**Review by aboodman — COMMENTED** [review]
lgtm

File: src/sync/mod.rs (hunk 1 above)
Comment (phritz) [reply src/sync/mod.rs:102]: thank you :)

**Pull request merged**